src/pages/ContentScripts/features/repo-networks/view.tsx

```diff
@@ -20,21 +21,21 @@ const graphStyle = {
 
 const View = ({ currentRepo, repoNetwork, developerNetwork }: Props): JSX.Element => {
   const [options, setOptions] = useState<HypercrxOptions>(defaults);
-
+  const { t, i18n } = useTranslation();
   useEffect(() => {
     (async function () {
       setOptions(await optionsStorage.getAll());
+      i18n.changeLanguage(options.locale);
     })();
-  }, []);
+  }, [options.locale]);
 
   return (
     <div>
       <div className="hypertrons-crx-border hypertrons-crx-container">
         <div className="hypertrons-crx-title">
-          <span>{getMessageByLocale('component_projectCorrelationNetwork_title', options.locale)}</span>
+          <span>{t('component_projectCorrelationNetwork_title')}</span>
           <div className="hypertrons-crx-title-extra">
-            {getMessageByLocale('global_period', options.locale)}: {REPO_PERIOD}{' '}
-            {getMessageByLocale('global_day', options.locale)}
+            {t('global_period')}: {REPO_PERIOD} {t('global_day')}
```

Comment:
Can we use [i18next Plurals](https://www.i18next.com/translation-function/plurals) here since `global_day` depends on a variable  `REPO_PERIOD`?